File: Java/Arrays/maximumAppearingElement.java
import java.util.*;

class maximumAppearingElement{

    public static void main(String [] args){
        Scanner sc = new Scanner(System.in);
        int n = 0;
        System.out.print("Enter the size of the arrays : ");
        n = sc.nextInt();
        ArrayList<Integer> arr1 = new ArrayList<>();
        ArrayList<Integer> arr2 = new ArrayList<>();
        System.out.println("Enter the upper and lower bounds of the arrays : ");
        for(int i=0;i<n;i++){
            int l = sc.nextInt();
            int r = sc.nextInt();
            arr1.add(l);
            arr2.add(r);
        }
        System.out.println("Maximum appearing element : " + maxElement(arr1, arr2));
    }

    public static int maxElement(ArrayList<Integer> arr1, ArrayList<Integer> arr2){
        int res = 0, n =  arr1.size();
        int [] arrSum = new int [100];
        for(int i=0;i<n;i++){
            arrSum[arr1.get(i)]++;
            arrSum[arr2.get(i) + 1]--;
        }
        ArrayList<Integer> pSum = new ArrayList<>();
        pSum.add(arrSum[0]);
        int curr = arrSum[0];
        for(int i=1;i<100;i++){
            int k = pSum.get(i-1) + arrSum[i];
            pSum.add(k);
            if(k > curr){
                curr = k;
                res = i;
            }
        }
        return res;
    }
};

File: Python/Arrays/maximumAppearingElement.py
def maxAppearingElement(arr1, arr2):
    arrPos = [0]*100
    for i in range(0, len(arr1)):
        arrPos[arr1[i]] = arrPos[arr1[i]] + 1
        arrPos[arr2[i] + 1] = arrPos[arr2[i] + 1] - 1
    pSum = list()
    pSum.append(arrPos[0])
    res = 0
    curr = arrPos[0]
    for i in range(1, len(arrPos)):
        k = pSum[i-1] + arrPos[i]
        pSum.append(k)
        if(k > curr):
            curr = k
            res = i
    return res

if __name__ == "__main__":
    n = int(input("Enter the size of the array : "))
    arr1 = list()
    arr2 = list()
    print("Enter the lower bounds and upper bounds of the array : ")
    for i in range(0, n):
        l = int(input())
        u = int(input())
        arr1.append(l)
        arr2.append(u)
    print("Maximum appearing element is : ", maxAppearingElement(arr1, arr2))